src/combinatorics/simple_graph/basic.lean

```diff
@@ -638,6 +648,51 @@ lemma delete_edges_eq_inter_edge_set (s : set (sym2 V)) :
   G.delete_edges s = G.delete_edges (s ∩ G.edge_set) :=
 by { ext, simp [imp_false] { contextual := tt } }
 
+lemma delete_edges_sdiff_eq_of_le {H : simple_graph V} (h : H ≤ G) :
+  G.delete_edges (G.edge_set \ H.edge_set) = H :=
+by { ext v w, split; simp [@h v w] { contextual := tt } }
+
+lemma edge_set_delete_edges (s : set (sym2 V)) :
+  (G.delete_edges s).edge_set = G.edge_set \ s :=
+by { ext e, refine sym2.ind _ e, simp }
+
+lemma edge_finset_delete_edges [fintype V] [decidable_eq V] [decidable_rel G.adj]
+  (s : finset (sym2 V)) [decidable_rel (G.delete_edges s).adj] :
+  (G.delete_edges s).edge_finset = G.edge_finset \ s :=
+by { ext e, simp [edge_set_delete_edges] }
+
+section delete_far
+variables (G) [ordered_ring 𝕜] [fintype V] [decidable_eq V] [decidable_rel G.adj]
+  {p : simple_graph V → Prop} {r r₁ r₂ : 𝕜}
+
+/-- A graph is `r`-*delete-far* from a property `p` if we must delete at least `r` edges from it to
+get a graph with the property `p`. -/
+def delete_far (p : simple_graph V → Prop) (r : 𝕜) : Prop :=
+∀ ⦃s⦄, s ⊆ G.edge_finset → p (G.delete_edges s) → r ≤ s.card
```

Comment:
It's a bit odd to let `r` be from an arbitrary ordered ring given that `s.card` is always a natural number, but I suppose the idea is that it's more convenient not having to coerce to a natural number, and furthermore that we can have additional lemmas to coerce `r` to a natural number as needed. In any cases, this seems like an interesting design experiment to introduce into mathlib.

Thanks!

bors r+

Reply:
The alternative would have been to use `ceil`. I didn't go for that because then all the API would have to be combined to the `floor_ring` API to get the actual results (for example we would need to chain monotonicity of `delete_verts` with monotonicity of `ceil`).